File: Sorting/Java/selectionSort.java
public class Main {
    public static void main(String[] args) {

        int arr [] = {8,3,4,2,10,22};

        selectionSort(arr);
        for(int x : arr){
            System.out.println(x);
        }

    }

    private static void selectionSort(int[] arr) {
        for(int i = 0;i<arr.length-1;i++){
            int min = i;

            for(int j = i;j<arr.length;j++){
                if(arr[j]<arr[min]){
                    min = j;
                }
            }

            //swap
            int temp = arr[i];
            arr[i] = arr[min];
            arr[min] = temp;

        }
    }
}

File: Sorting/C++/selectionSort.cpp
#include<iostream>
using namespace std;


void selectionSort(int arr[],int n){

  for(int i = 0;i<n-1;i++){
    int min = i;
    for(int j = i;j<n;j++){
      if(arr[j]<arr[min]){
        min = j;
      }
    }

    //swapping 
    int temp = arr[min];
    arr[min] = arr[i];
    arr[i] = temp;
  }
}
int main(){
  int n ;
  cout<<"Enter the size of the array ";
  cin>>n;

  int arr[n];
  cout<<"Enter the Elements : ";
  for(int i=0;i<n;i++){
    cin>>arr[i];
  }

  selectionSort(arr,n);
  cout<<"Printing Sorted Elements : ";
  for(int i=0;i<n;i++){
    cout<<arr[i]<<" ";
  }
  return 0;
}
